### DISCUSSION TITLE
Help: Mis-matched ISA version for 'zicsr' extension

### DISCUSSION THREAD

**Discussion by pratipm** [stnolting/neorv32#630] [Q&A]
Please help me spot my mistake. I have a brand new install of risc-v gcc toolchain from https://github.com/stnolting/riscv-gcc-prebuilt and a brand new install of neorv32 from github on Ubuntu 22.04.
While compiling any of the example programs, like hello_world with make clean exe, getting umpteen number of errors:

/ssd/pratip/riscv/bin/../lib/gcc/riscv32-unknown-elf/10.2.0/../../../../riscv32-unknown-elf/bin/ld: error: ../../../sw/common/crt0.S.o: Mis-matched ISA version for 'zicsr' extension. 0.0 vs 2.0
/ssd/pratip/riscv/bin/../lib/gcc/riscv32-unknown-elf/10.2.0/../../../../riscv32-unknown-elf/bin/ld: failed to merge target specific data of file ../../../sw/common/crt0.S.o
/ssd/pratip/riscv/bin/../lib/gcc/riscv32-unknown-elf/10.2.0/../../../../riscv32-unknown-elf/bin/ld: -march=: ISA string must begin with rv32 or rv64
/ssd/pratip/riscv/bin/../lib/gcc/riscv32-unknown-elf/10.2.0/../../../../riscv32-unknown-elf/bin/ld: failed to merge target specific data of file /ssd/pratip/riscv/bin/../lib/gcc/riscv32-unknown-elf/10.2.0/../../../../riscv32-unknown-elf/lib/libc.a(lib_a-memcpy.o)
/ssd/pratip/riscv/bin/../lib/gcc/riscv32-unknown-elf/10.2.0/../../../../riscv32-unknown-elf/bin/ld: -march=: ISA string must begin with rv32 or rv64

Thanks

**Comment by pratipm** [comment]
Found another copy of riscv toolchain on my machine as part of Altera/Intel tools. The riscv gcc version is 11.2. With that toolset, I can compile and generate exe with just a warning from ld:
cannot find default versions of the ISA extension `zicsr'

**Reply by stnolting** [reply]
Hey there!

> `/ssd/pratip/riscv/bin/../lib/gcc/riscv32-unknown-elf/10.2.0`

This does not look like the toolchain from  https://github.com/stnolting/riscv-gcc-prebuilt - the newest version there is using GCC version 12.1 :wink: So maybe there is a problem with your `PATH` variable.

For the records: the newer RISC-V GCC toolchains require `zicsr` to be present in the architecture string (e.g. `MARCH=rv32i_zicsr`) if you are using any CSR instructions.
